drivers/classic/src/calc_air_temperature.c

```diff
@@ -176,48 +174,56 @@ HourlyT(int     Dt,
 
 /******************************************************************************
  * @brief    This function estimates the times of minimum and maximum
- *           temperature for each day of the simulation, based on the hourly
+ *           temperature for each day of the simulation, based on the diurnal
  *           cycle of incoming solar radiation.
  *****************************************************************************/
 void
-set_max_min_hour(double *hourlyrad,
-                 int     ndays,
-                 int    *tmaxhour,
-                 int    *tminhour)
+set_max_min_sec(double *subdailyrad,
+                int     ndays,
+                int     stepsize,
+                int    *tmaxsec,
+                int    *tminsec)
 {
-    int risehour;
-    int sethour;
-    int hour;
-    int i;
+    size_t step;
+    int    risesec;
+    int    setsec;
+    int    sec;
+    int    i;
+    size_t stepsperday;
+
+    stepsperday = SEC_PER_DAY / stepsize;
 
     for (i = 0; i < ndays; i++) {
-        risehour = MISSING;
-        sethour = MISSING;
-        for (hour = 0; hour < 12; hour++) {
-            if (hourlyrad[i * HOURS_PER_DAY + hour] > 0 &&
-                (i * HOURS_PER_DAY + hour == 0 ||
-                 hourlyrad[i * HOURS_PER_DAY + hour - 1] <= 0)) {
-                risehour = hour;
+        risesec = MISSING;
+        setsec = MISSING;
+        for (step = 0, sec = 0;
+             sec < 12 * SEC_PER_HOUR;
+             step++, sec += stepsize) {
+            if (subdailyrad[i * SEC_PER_DAY + step] > 0 &&
+                (i * stepsperday + step == 0 ||
+                 subdailyrad[i * stepsperday + step - 1] <= 0)) {
+                risesec = sec;
             }
         }
-        for (hour = 12; hour < HOURS_PER_DAY; hour++) {
-            if (hourlyrad[i * HOURS_PER_DAY + hour] <= 0 &&
-                hourlyrad[i * HOURS_PER_DAY + hour - 1] >
-                0) {
-                sethour = hour;
+        for (step = step + 1, sec = 12 * SEC_PER_HOUR;
+             sec < SEC_PER_DAY;
+             step++, sec += stepsize) {
+            if (subdailyrad[i * stepsperday + sec] <= 0 &&
+                subdailyrad[i * stepsperday + sec - 1] > 0) {
+                setsec = sec;
             }
         }
-        if (i == ndays - 1 && sethour == MISSING) {
-            sethour = 23;
+        if (i == ndays - 1 && setsec == MISSING) {
+            setsec = 23 * SEC_PER_HOUR;
         }
-        if (risehour >= 0 && sethour >= 0) {
-            tmaxhour[i] = 0.67 * (sethour - risehour) + risehour;
-            tminhour[i] = risehour - 1;
+        if (risesec >= 0 && setsec >= 0) {
+            tmaxsec[i] = 0.67 * (setsec - risesec) + risesec;
+            tminsec[i] = risesec - 1;
```

Comment:
This function does not have a citation.  @bartnijssen or @tbohn, do you know where we could check on the intended implementation of this.  If `Tmin` is intended to occur 1hr before sunrise, then the code should read `tminsec[i] = risesec - SEC_PER_HOUR;`, if it is intended to occur in the previous timestep, it would read `tminsec[i] = risesec - stepsize;`.  

I think we also need to check that `(risesec - (some quantity) > 0;` since this wasn't possible before.